specs/exponentiation-circuit.md

```diff
@@ -0,0 +1,254 @@
+# Exponentiation Circuit
```

Comment:
Can we indicate this as a design doc in both title and the filename? So that people don't confuse the difference between `spec/exp-proof.md` and `spec/exponentiation-circuit.md`
```suggestion
# Exponentiation Proof Design Document.
```